### PR TITLE
Create klusterlet addons on managed clusters

### LINKED ISSUE #7903: On MDR deployments, importing cluster with ACM does not create klusterlet addons
This section needs to be configured for cli way: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/clusters/cluster_mce_overview#importing-cluster-manual

### PR DESCRIPTION
**Pull request by parikshithb**
fixes #7903

### CODE REVIEW

File: ocs_ci/ocs/acm/acm.py [hunk 1]
```diff
@@ -513,6 +514,20 @@ def import_clusters_via_cli(clusters):
...
+
+        log.info("Create the Klusterlet addon configuration")
+        klusterlet_config = templating.load_yaml(constants.ACM_HUB_KLUSTERLET_YAML)
+        klusterlet_config["metadata"]["name"] = cluster[0]
+        klusterlet_config["metadata"]["namespace"] = cluster[0]
+        klusterlet_config_obj = OCS(**klusterlet_config)
+        klusterlet_config_obj.apply(**klusterlet_config)
+
+        log.info("Wait for up to 60s for addon pods to be initialized")
+        time.sleep(60)
```
Comment (shylesh) [inline ocs_ci/ocs/acm/acm.py:526]: @parikshithb , Instead of sleep can we wait on the resource with a state ?

File: ocs_ci/ocs/acm/acm.py [hunk 2]
```diff
@@ -513,6 +514,20 @@ def import_clusters_via_cli(clusters):
             column="JOINED",
             resource_name=cluster[0],
         )
+
+        log.info("Create the Klusterlet addon configuration")
+        klusterlet_config = templating.load_yaml(constants.ACM_HUB_KLUSTERLET_YAML)
+        klusterlet_config["metadata"]["name"] = cluster[0]
+        klusterlet_config["metadata"]["namespace"] = cluster[0]
+        klusterlet_config_obj = OCS(**klusterlet_config)
+        klusterlet_config_obj.apply(**klusterlet_config)
```
Comment (shylesh) [inline ocs_ci/ocs/acm/acm.py:523]: are we editing the existing resource OR creating a new one ?

**Review by shylesh — COMMENTED** [review]
few minor comments, otherwise looks good.

File: ocs_ci/ocs/acm/acm.py (hunk 1 above)
Comment (parikshithb) [reply ocs_ci/ocs/acm/acm.py:526]: Creating klusterlet yaml, creates addon pods in "open-cluster-management-agent-addon", waiting here for these pods to be in initializing/ContainerCreating before validating all of them to be in running state by calling wait_for_pods_to_be_running.

File: ocs_ci/ocs/acm/acm.py (hunk 2 above)
Comment (parikshithb) [reply ocs_ci/ocs/acm/acm.py]: Creating new, made the change.

File: ocs_ci/ocs/acm/acm.py (hunk 1 above)
Comment (ebenahar) [reply ocs_ci/ocs/acm/acm.py:526]: @parikshithb why is sleep needed in that case? Should we add that time to the timeout we pass to wait_for_pods_to_be_running()?

Comment (shylesh) [reply ocs_ci/ocs/acm/acm.py:526]: > @parikshithb why is sleep needed in that case? Should we add that time to the timeout we pass to wait_for_pods_to_be_running()?

I agree with @ebenahar , we should pass the timeout while waiting for pods.

**Comment by parikshithb** [comment]
Removed sleep and validated here: http://pastebin.test.redhat.com/1104396

File: ocs_ci/ocs/acm/acm.py (hunk 1 above)
Comment (parikshithb) [reply ocs_ci/ocs/acm/acm.py]: Agree, made the change and validated.

**Review by Akarsha-rai — APPROVED** [review]
lgtm

**Review by ebenahar — APPROVED** [review]

**Pull request merged**